internal/pkg/api/hardware.go

```diff
@@ -43,3 +37,57 @@ func NewHardwareSlice(csvFile string) ([]*Hardware, error) {
 
 	return hardware, nil
 }
+
+func NewHardwareMap(csvFile string) (map[string]*Hardware, error) {
+	slice, err := NewHardwareSlice(csvFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create hardware map from hardware csv file: %v", err)
+	}
+	return HardwareSliceToMap(slice), nil
+}
+
+func HardwareSliceToMap(slice []*Hardware) map[string]*Hardware {
```

Comment:
The function could do with a code comment indicating the behavior when 2 hardware with the same ID but different data are supplied in the slice. The behavior you've got is to use the first instance. 

In general, comment exported functions. Even if its intended for use in test because its still just code.